docs/repo/dependencies-node-roadmap.md

```diff
@@ -36,66 +36,71 @@ In general, items _directly_ referenced by the project file (e.g., through `Refe
+As evaluations and design-time builds occur, CPS pushes project changes through TPL Dataflow blocks (available via `IProjectSubscriptionService`, where `ProjectRuleSource` provides evaluation data, and `JointRuleSource` provides design-time build data).
 
-These handlers add the added and removed items to a [`DependenciesRuleChangeContext`][20]. This fires its `DependenciesChanged` event and passes along the [`DependenciesRuleChangeContext`][20] to event subscribers.
+Various implementations of [`IDependenciesRuleHandler`][IDependenciesRuleHandler] exist, and each specifies the set of rules they wish to handle (e.g. `PackageReference`, `ResolvedProjectReference`, etc.). The abstract class [`DependenciesRuleHandlerBase`][DependenciesRuleHandlerBase] exists to make implementing [`IDependenciesRuleHandler`][IDependenciesRuleHandler] easier.
 
-Each project has an instance of [`DependencySubscriptionsHost`][7] implementing [`IDependenciesSnapshotProvider`][2]. It subscribes to this `DependenciesChanged` event, ultimately handling it in `UpdateDependenciesSnapshotAsync`. If there are relevant changes it creates a new [`DependenciesSnapshot`][21] and fires its own `SnapshotChanged` event.
+The [`DependencyRulesSubscriber`][DependencyRulesSubscriber] (implementing [`IDependencyCrossTargetSubscriber`][IDependencyCrossTargetSubscriber]) subscribes via Dataflow to the union of rules specified by the handlers. When updates are received, a [`CrossTargetDependenciesChangesBuilder`][CrossTargetDependenciesChangesBuilder] is instantiated and each handler is given a chance to add/update/remove `IDependencyModel` instances through the builder. Once complete, the `IDependencyCrossTargetSubscriber.DependenciesChanged` event is fired, carrying dependency model changes.
 
-The [`DependenciesProjectTreeProvider`][1] subscribes to this event and handles updating the tree.
+Each project has an instance of [`DependencySubscriptionsHost`][DependencySubscriptionsHost] (implementing [`IDependenciesSnapshotProvider`][IDependenciesSnapshotProvider]) that holds the latest `DependenciesSnapshot` object. It imports `IDependencyCrossTargetSubscriber` implementations (such as `DependencyRulesSubscriber`) and subscribes to their `DependenciesChanged` events. When these events fire, the current snapshot is combined with changes to produce a new snapshot. That snapshot is then propagated via the `IDependenciesSnapshotProvider.SnapshotChanged` event.
```

Comment:
[`DependencySubscriptionsHost`][DependencySubscriptionsHost]
should be
[`DependenciesSnapshotProvider`][DependenciesSnapshotProvider]